pkg/server/proto/server.proto

```diff
@@ -4749,6 +4759,28 @@ message Pipeline {
   }
 }
 
+message PipelineRun {
+  string id = 1;
+  // The sequence number for this pipeline run.
+  uint64 sequence = 2;
+
+  Ref.Pipeline pipeline = 3;
+
+  repeated Ref.Job jobs = 4;
+
```

Comment:
> Yes, I'd like to put this in the CLI PR coming though, since there'll be other changes with the PipelineBundle, I want to think through it all together.

That sounds like the best approach for sure 👍 

> Top of mind, there is a problem down the line with concurrent pipeline steps, and two concurrent jobs... which one do we pick as the complete time?

Oh wow, yeah that’s a great point, and more reason for the server to determine that kind of info.